packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-results-header.tsx

```diff
@@ -1,44 +1,69 @@
 import React from 'react';
-import { css, spacing } from '@mongodb-js/compass-components';
+import { css, Overline, spacing } from '@mongodb-js/compass-components';
 import type { ResultsViewType } from './pipeline-results-list';
 import PipelinePagination from './pipeline-pagination';
 import PipelineResultsViewControls from './pipeline-results-view-controls';
 import { connect } from 'react-redux';
 import type { RootState } from '../../modules';
 import { changeViewType } from '../../modules/aggregation';
 import { getStageOperator, isOutputStage } from '../../utils/stage';
+import { PipelineOutputOptionsMenu } from '../pipeline-output-options-menu';
+import type { PipelineOutputOption } from '../pipeline-output-options-menu';
 
 type PipelineResultsHeaderProps = {
   onChangeResultsView: (viewType: ResultsViewType) => void;
   resultsViewType: ResultsViewType;
   isMergeOrOutPipeline: boolean;
+  onChangePipelineOutputOption: (val: PipelineOutputOption) => void;
+  pipelineOutputOption: PipelineOutputOption;
 };
 
-const controlsStyles = css({
+const containerStyles = css({
   display: 'flex',
-  gap: spacing[2],
-  justifyContent: 'flex-end',
+  justifyContent: 'space-between',
   alignItems: 'center',
 });
 
-export const PipelineResultsHeader: React.FunctionComponent<PipelineResultsHeaderProps> =
-  ({ onChangeResultsView, resultsViewType, isMergeOrOutPipeline }) => {
-    if (isMergeOrOutPipeline) {
-      return null;
-    }
-
-    return (
-      <div data-testid="pipeline-results-header">
-        <div className={controlsStyles}>
-          <PipelinePagination />
-          <PipelineResultsViewControls
-            value={resultsViewType}
-            onChange={onChangeResultsView}
+const groupStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+});
+
+export const PipelineResultsHeader: React.FunctionComponent<
+  PipelineResultsHeaderProps
+> = ({
+  onChangeResultsView,
+  resultsViewType,
+  isMergeOrOutPipeline,
+  onChangePipelineOutputOption,
+  pipelineOutputOption,
+}) => {
+  if (isMergeOrOutPipeline) {
+    return null;
+  }
+  return (
+    <div className={containerStyles} data-testid="pipeline-results-header">
+      {process?.env?.COMPASS_ENABLE_AS_TEXT_PIPELINE === 'true' ? (
+        <div className={groupStyles}>
+          <Overline>All Results</Overline>
+          <PipelineOutputOptionsMenu
+            option={pipelineOutputOption}
+            onChangeOption={onChangePipelineOutputOption}
           />
         </div>
+      ) : (
+        <div />
```

Comment:
The usual pattern in react for this `{condition $$ <Component />}`, you don't need to render a fallback if fallback is nothing